### PR TITLE
Finished 5.2 - piece of cake exercise

### PR DESCRIPTION
**Pull request by EitanAb8**

### CODE REVIEW

File: piece_of_cake.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+def get_recipe_price(prices, optionals=None, **ingredients):
+    # This function receives:
```
Comment (talbarfm) [inline piece_of_cake.py]: Capture

File: piece_of_cake.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    # prices - dictionary of ingredients and their price for 100 grams.
+    # optionals - ingredients to ignore in the cart
+    # ingredients - grams for each ingredient we want to buy
+    # the function will calculate and return the sum of the ingredients price
+
+    final_price = 0
+    for name, price in prices.items():
+        final_price += price*(ingredients[name]/100) if name in ingredients.keys() else 0
```
Comment (talbarfm) [inline piece_of_cake.py]: space around operators

Comment (talbarfm) [reply piece_of_cake.py]: ingredients[name]/100 makes less sense then price/100 - you want the cost per gram multiplied by the amount of grams, not the cost per a 100 grams multiplies by the amount/100, even though the result is the same

File: piece_of_cake.py (hunk 1 above)
Comment (talbarfm) [reply piece_of_cake.py]: document in this format

File: piece_of_cake.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+def get_recipe_price(prices, optionals=None, **ingredients):
+    # This function receives:
+    # prices - dictionary of ingredients and their price for 100 grams.
+    # optionals - ingredients to ignore in the cart
+    # ingredients - grams for each ingredient we want to buy
+    # the function will calculate and return the sum of the ingredients price
+
+    final_price = 0
+    for name, price in prices.items():
```
Comment (talbarfm) [inline piece_of_cake.py]: replace this for loop with reduce function from itertools

File: piece_of_cake.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        final_price += price*(ingredients[name]/100) if name in ingredients.keys() else 0
+    return final_price
+
+
+def main():
+
```
Comment (talbarfm) [inline piece_of_cake.py]: you dont need this newline

**Review by talbarfm — CHANGES_REQUESTED** [review]

**Review by lovodkin93 — APPROVED** [review]
good job!

**Pull request merged**